internal/ethapi/api_test.go

```diff
@@ -2062,3 +2067,102 @@ func testRPCResponseWithFile(t *testing.T, testid int, result interface{}, rpc s
 	}
 	require.JSONEqf(t, string(want), string(data), "test %d: json not match, want: %s, have: %s", testid, string(want), string(data))
 }
+
+func TestBlockChainAPI_isAllowedToBeQueried(t *testing.T) {
+	t.Parallel()
```

Comment:
nit I'm a `t.Parallel()` maniac. This with the test race detector did save me from time to time.